docs/DeveloperGuide.md

```diff
@@ -1,29 +1,240 @@
+    - **Process**:
+        - Adds the `transaction` to the `transactions` list.
+        - Updates the `invertedIndex` to include the new `transaction` for quick search.
+        - Sorts the `transactions` list by date using a custom comparator.
+
+2. **deleteTransaction(index : int) : Transaction**
+    - **Parameters**:
+        - `index`: The position of the `Transaction` object to be removed.
+    - **Process**:
+        - Removes the `Transaction` at the specified `index` from `transactions`.
+        - Updates the `invertedIndex` to reflect the deletion.
+        - Returns the removed `Transaction`.
+
+3. **getTransactions() : ArrayList<Transaction>**
+    - **Parameters**: None
+    - **Process**:
+        - Returns the complete list of `Transaction` objects in `transactions`.
+
+4. **searchTransactionsByKeywords(keywords : List<String>) : List<Transaction>**
+    - **Parameters**:
+        - `keywords`: A list of keywords to search for within transaction descriptions.
+    - **Process**:
+        - Looks up each `keyword` in the `invertedIndex` to find matching transactions.
+        - Aggregates and counts relevance for each match.
+        - Sorts the results by relevance and returns the list of matched transactions.
+![SearchByKeywords](./diagrams/TransactionList/transactionlist-class-SearchKeyword-diagram.png)
+5. **getExpensesByCategory(category : Category) : List<Transaction>**
+    - **Parameters**:
+        - `category`: The `Category` to filter expenses by.
+    - **Process**:
+        - Filters `transactions` to include only `Expense` objects with the specified `category`.
+        - Returns the filtered list of expenses.
+![SearchByCategory](./diagrams/TransactionList/transactionlist-Sequence-SearchByCategoty-diagram.png)
+
+### AddIncomeCommand
+The `AddIncomeCommand` class handles the logic for adding an income transaction to the `TransactionList` by parsing input arguments, creating a new `Income` instance, and updating the transaction list.
+
+![AddIncomeCommand](./diagrams/addincomecommand/addincomecommand-class-diagram.png)
+
+### Class Responsibilities
+
+1. **Command Parsing and Validation**: The class validates required fields (e.g., amount) and parses optional fields (e.g., date).
+2. **Transaction Creation**: An `Income` transaction is instantiated and initialized with a description, amount, and date.
+3. **Transaction Storage**: Upon creation, the transaction is saved to the `TransactionList` and persisted using `Storage`.
+
+### Class Attributes
+1. **COMMAND_WORD**: `String`
+    - **Description**: The keyword triggering this command.
+2. **COMMAND_MANDATORY_KEYWORDS**: `String[]`
+    - **Description**: Array storing mandatory argument keywords.
+3. **COMMAND_EXTRA_KEYWORDS**: `String[]`
+    - **Description**: Array storing optional argument keywords.
+4. **transactions**: `TransactionList`
+    - **Description**: Stores the current list of all transactions.
+
+### Class Main Methods
+
+1. **execute()**
```

Comment:
i think you are missing the return type of the command based on the UML notation